nodes/modifier_make/subdivide_mk2.py

```diff
@@ -0,0 +1,399 @@
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_common(context, layout)
+        self.draw_options(context, layout)
+        layout.label(text="Ouput Numpy:")
+        r = layout.row()
+        for i in range(4):
+            r.prop(self, "out_np", index=i, text=socket_names[i], toggle=True)
+
+    def rclick_menu(self, context, layout):
+        '''right click sv_menu items'''
+        layout.prop_menu_enum(self, "falloff_type")
+        layout.prop_menu_enum(self, "corner_type")
+        layout.prop(self, "grid_fill", toggle=True)
+        layout.prop(self, "single_edge", toggle=True)
+
+
+        layout.prop(self, "only_quads", toggle=True)
+        layout.prop(self, "smooth_even", toggle=True)
+
+        layout.label(text="Ouput Numpy:")
+
+        for i in range(4):
+            layout.prop(self, "out_np", index=i, text=socket_names[i], toggle=True)
+    def sv_init(self, context):
+        inew = self.inputs.new
+        inew('SvVerticesSocket', "Vertices")
+        inew('SvStringsSocket', 'Edges')
+        inew('SvStringsSocket', 'Faces')
+        inew('SvStringsSocket', 'FaceData')
+        inew('SvStringsSocket', 'EdgeMask')
+
+        inew('SvStringsSocket', 'Cuts').prop_name = "cuts"
+        inew('SvStringsSocket', 'Smooth').prop_name = "smooth"
+        inew('SvStringsSocket', 'Fractal').prop_name = "fractal"
+        inew('SvStringsSocket', 'AlongNormal').prop_name = "along_normal"
+        inew('SvStringsSocket', 'Seed').prop_name = "seed"
+
+        onew = self.outputs.new
+        onew('SvVerticesSocket', 'Vertices')
+        onew('SvStringsSocket', 'Edges')
+        onew('SvStringsSocket', 'Faces')
+        onew('SvStringsSocket', 'FaceData')
+
+        onew('SvVerticesSocket', 'NewVertices')
+        onew('SvStringsSocket', 'NewEdges')
+        onew('SvStringsSocket', 'NewFaces')
+
+        onew('SvVerticesSocket', 'OldVertices')
+        onew('SvStringsSocket', 'OldEdges')
+        onew('SvStringsSocket', 'OldFaces')
+
+        self.update_mode(context)
+
+    def mute_node(self):
+        origin_socket = [0, 1, 2, 3, 0, 1, 2, 0, 1, 2]
+
+        for i, s in zip(origin_socket, self.outputs):
+            if self.inputs[i].is_linked:
+                s.sv_set(self.inputs[i].sv_get(deepcopy=False))
+
+    def freeze_node(self):
+        if 'storage' in self.mem.keys():
+            for m, s in zip(self.mem['storage'], self.outputs):
+                s.sv_set(m)
+            return True
+        return False
+
+    def get_result_pydata(self, geom):
+        new_verts = [v for v in geom if isinstance(v, BMVert)]
+        new_edges = [e for e in geom if isinstance(e, BMEdge)]
+        new_faces = [f for f in geom if isinstance(f, BMFace)]
+
+        new_verts = [tuple(v.co) for v in new_verts]
+        new_edges = [[edge.verts[0].index, edge.verts[1].index] for edge in new_edges]
+        new_faces = [[v.index for v in face.verts] for face in new_faces]
+
+        return new_verts, new_edges, new_faces
```

Comment:
This method looks like should be separate function or should be the method of some other class.

Reply:
moved to the sv_bmesh_utils